File: IT/Programmazione funzionale/Esercizi/Python/esercizio17.py
# file esercizio17.py

from __future__ import print_function, division
from random import randint
from sys import version_info

if version_info[0] > 2: from functools import reduce

def avg(li):
	return reduce(lambda a, b : a + b, li) / len(li)

li = [randint(1, 100) for n in range(25)]

m = avg(li)

print(li)
print("Media: {}".format(m))

File: IT/2/Programmazione funzionale/Esercizi/Java/Esercizio17.java
/* file Esercizio17.java */

import java.util.Locale;
import java.util.List;
import java.util.Random;
import java.util.stream.Collectors;

public class Esercizio17 {
	private static double avg(List<Integer> li) {
		return (double) li.stream().reduce(0, (m, n) -> m + n) / (double) li.size();
	}

	public static void main(String [] args) {
		Random rand = new Random();
		List<Integer> li = rand.ints(25, 1, 100).boxed().collect(Collectors.toList());

		double m = avg(li);

		System.out.println(li);
		System.out.println("Media: " + String.format(Locale.ROOT, "%.2f", m) + ".");
	}
}

